src/compiler/checker.ts

```diff
@@ -138,6 +138,11 @@ namespace ts {
         const silentNeverType = createIntrinsicType(TypeFlags.Never, "never");
 
         const emptyObjectType = createAnonymousType(undefined, emptySymbols, emptyArray, emptyArray, undefined, undefined);
+
+        const emptyTypeLiteralSymbol = createSymbol(SymbolFlags.TypeLiteral | SymbolFlags.Transient | SymbolFlags.Synthesized, "empty");

```

Comment:
Why are you marking this as `SymbolFlags.Synthesized`? We use that flag for union and intersection properties. Also, I would use the name `"__type"`, that's what we do in the binder.